node/src/restarter.rs

```diff
@@ -0,0 +1,126 @@
+            let mut store_path = storage_base_path.clone();
+            store_path.push(format!("epoch{}", committee.epoch()));
+            let store = NodeStorage::reopen(store_path);
+
+            // Restart the relevant components.
+            let primary_handles = Node::spawn_primary(
+                keypair,
+                Arc::new(ArcSwap::new(Arc::new(committee.clone()))),
+                &store,
+                parameters.clone(),
+                /* consensus */ true,
+                execution_state.clone(),
+                tx_output.clone(),
+                &Registry::new(),
+            )
+            .await
+            .unwrap();
+
+            let worker_handles = Node::spawn_workers(
+                name.clone(),
+                /* worker_ids */ vec![0],
+                Arc::new(ArcSwap::new(Arc::new(committee.clone()))),
+                &store,
+                parameters.clone(),
+                &Registry::new(),
+            );
+
+            handles.extend(primary_handles);
+            handles.extend(worker_handles);
+
+            // Wait for a committee change.
+            let (new_keypair, new_committee) = match rx_reconfigure.recv().await {
+                Some(x) => x,
+                None => break,
+            };
+            tracing::info!("Starting reconfiguration with committee {committee}");
+
+            // Shutdown all relevant components.
+            let address = committee
+                .primary(&name)
+                .expect("Our key is not in the committee")
+                .worker_to_primary;
+            let message =
+                WorkerPrimaryMessage::<PublicKey>::Reconfigure(ReconfigureNotification::Shutdown);
+            let primary_cancel_handle = primary_network.send(address, &message).await;
+
+            let addresses = committee
+                .our_workers(&name)
+                .expect("Our key is not in the committee")
+                .into_iter()
+                .map(|x| x.primary_to_worker)
+                .collect();
+            let message =
+                PrimaryWorkerMessage::<PublicKey>::Reconfigure(ReconfigureNotification::Shutdown);
+            let worker_cancel_handles = worker_network.broadcast(addresses, &message).await;
+
+            // Ensure the message has been received.
+            primary_cancel_handle
+                .await
+                .expect("Failed to notify primary");
+            join_all(worker_cancel_handles).await;
+            tracing::debug!("Committee reconfiguration message successfully sent");
+
+            // Wait for the components to shut down.
+            join_all(handles.drain(..)).await;
+            tracing::debug!("All tasks successfully exited");
+
+            // Give it an extra second in case the last task to exit is a network server. The OS
+            // may need a moment to make the TCP ports available again.
+            tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
```

Comment:
There is a better method for doing this in the network crate if you know which ports you'll claim. Do you?

Reply:
I don't really know which port that is sadly, it can be any of them (on the Primary or any worker)

Reply:
what is the method?